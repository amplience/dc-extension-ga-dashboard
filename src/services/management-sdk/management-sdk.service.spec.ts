import { HttpClient } from "dc-extensions-sdk";
import type { ClientConnection } from "message-event-channel";
import getManagementClient from "./management-sdk.service";
import { DynamicContent } from 'dc-management-sdk-js';

jest.mock('dc-management-sdk-js');

describe('ManagementSDK - getManagementClient', () => {
  it('should return a new client', async () => {
    getManagementClient(new HttpClient({} as ClientConnection));
    expect((DynamicContent as jest.Mock).mock.calls).toMatchSnapshot();
  });
});
