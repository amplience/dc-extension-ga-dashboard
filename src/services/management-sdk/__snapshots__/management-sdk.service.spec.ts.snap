// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagementSDK - getManagementClient should return a new client 1`] = `
Array [
  Array [
    Object {
      "client_id": "",
      "client_secret": "",
    },
    Object {},
    HttpClient {
      "DEFAULT_ERROR": Object {
        "data": Object {
          "errors": Array [
            Object {
              "code": "UNKNOWN",
              "level": "ERROR",
              "message": "Unknown error",
            },
          ],
        },
        "status": 403,
      },
      "connection": Object {},
    },
  ],
]
`;
